pkg/querier/blocks_store_queryable.go

```diff
@@ -879,6 +884,18 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(ctx context.Context, sp *stor
 				}
 			}
 
+			// debug
+			var chunkInfo *chunkreplyformatter.ChunkReplyFormatter
+			if debugContinuousTest {
+				for _, cm := range convertedMatchers {
+					if cm.Name == "__name__" && cm.Value == "mimir_continuous_test_sine_wave_v2" {
+						chunkInfo = chunkreplyformatter.NewChunkReplyFormatter()
+						break
+					}
+				}
+			}
```

Comment:
Rather than specifically checking for the continuous-test series, what if we relied on continuous-test setting a HTTP header (eg. `Mimir-Enable-Query-Debugging: true`) when it makes the query request that enables this logging?

Then we could reuse the same logging for debugging other queries as well if need be.

Reply:
yep, I started from what we had now. We can add it incrementally . I had some wild ideas, but it's a scope creep :)